packages/jsii/lib/project-info.ts

```diff
@@ -312,9 +312,12 @@ class DependencyResolver {
 
       const actualVersion = resolved.dependencyInfo.assembly.version;
       if (!semver.satisfies(actualVersion, declaration)) {
-        throw new Error(
-          `Declared dependency on version ${declaration} of ${name}, but version ${actualVersion} was found`,
-        );
+        // Allow URI declarations
+        if (!declaration.match(/^[a-z]+:/)) {
```

Comment:
What gets recorded in the output `.jsii` assembly in this case? Does this produce correct dependency declarations in other languages once consumed by `jsii-pacmak`?

Reply:
That is an excellent question. Where would I look for find that?

Reply:
On a project that uses this feature, you'd look at the `dependencies` and `dependencyClosure` keys within the `.jsii` file created as a result of compilation...

And then you'd generate code with `jsii-pacmak --code-only` and inspect dependency declarations in `pom.xml` for Java, the `.csproj` file for .NET, the `go.mod` file for Go, or the `setup.py` file for Python (reviewing one of these should be enough - they should all have comparable declarations).

I'm trying to think of a way we can have a test for this as part of the codebase, too... I think we could change one of `@scope/jsii-calc-*` packages to use a `file://` URI specifier and that hopefully does not break the mono repo structure we have today...